examples/api/lib/material/theme/theme_extension.1..dart

```diff
@@ -0,0 +1,134 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for ThemeExtension
+
+import 'package:flutter/material.dart';
+
+@immutable
+class MyColors implements ThemeExtension {
+  const MyColors({
+    this.blue,
+    this.red,
+  });
+
+  final Color? blue;
+  final Color? red;
+
+  @override
+  MyColors copyWith({Color? red, Color? blue}) {
+    return MyColors(
+      blue: blue ?? this.blue,
+      red: red ?? this.red,
+    );
+  }
+
+  @override
+  MyColors lerp(ThemeExtension? other, double t) {
+    if (other is MyColors) {
+      return MyColors(
+        blue: Color.lerp(blue, other.blue, t),
+        red: Color.lerp(red, other.red, t),
+      );
+    } else {
+      return this;
+    }
+  }
+
+  // Optional
+  @override
+  bool operator ==(Object other) {
+    return other is MyColors &&
+        other.blue == blue &&
+        other.red == red;
+  }
+
+  // Optional
+  @override
+  int get hashCode {
+    return hashList(<Object?>[
+      blue,
+      red,
+    ]);
+  }
+}
+
+extension on ThemeData {
+  MyColors get myColors => themeExtension! as MyColors;
+}
+
+
+void main() => runApp(const MyApp());
+
+class MyApp extends StatefulWidget {
+  const MyApp({Key? key}) : super(key: key);
+
+  static const String _title = 'Flutter Code Sample';
```

Comment:
The title boilerplate isn't really necessary. If a title is really worthwhile, it's just as well specified with `title: 'my title'` in the MaterialApp ctor.

Reply:
I'm following the convention set in all other code samples